operator/auth/spire/client.go

```diff
@@ -152,7 +172,12 @@ func (c *Client) connect(ctx context.Context) (*grpc.ClientConn, error) {
 	}
 
 	tlsConfig := tlsconfig.MTLSClientConfig(source, source, tlsconfig.AuthorizeMemberOf(trustedDomain))
-	conn, err := grpc.Dial(c.cfg.SpireServerAddress, grpc.WithTransportCredentials(credentials.NewTLS(tlsConfig)))
+
+	c.log.WithFields(map[string]interface{}{
+		"spire_service_address": c.cfg.SpireServerAddress,
+		"resolved_address":      resolvedTarget,
+	}).Info("Trying to connect to SPIRE server")
+	conn, err := grpc.Dial(*resolvedTarget, grpc.WithTransportCredentials(credentials.NewTLS(tlsConfig)))
 	if err != nil {
 		return nil, fmt.Errorf("failed to create connection to SPIRE server: %w", err)
 	}
```

Comment:
Nit: Might be better to log after a successful connection. Also, please use `pkg/logging/logfields` for the fields. Some are already defined that we can reuse.